web/server.go

```diff
@@ -197,13 +197,19 @@ func (s *Server) serveHealthz(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) serveSearch(w http.ResponseWriter, r *http.Request) {
-	result, err := s.serveSearchErr(r)
+	qvals := r.URL.Query()
+
+	debugScore := false
+	if qvals.Get("debug") == "1" {
+		debugScore = true
+	}
```

Comment:
should this be done in serveSearchErr like the rest of the query param parsing?

Reply:
I think both is fine. We already parse the query params for the JSON API and the caller `serveSearch` needs to know whether the debug info should be displayed. However, I tried it out and moved `debugScore` to `serveSearchErr` which made it possible to get rid of the new options struct and the helper type I previously introduced, which is definitely a plus. The template now decides to render the debug info based on whether the `ScoreDebug` string is empty or not, which is not as explict as before but I think it is fine.